src/components/smart_objects/include/smart_objects/number_schema_item.h

```diff
@@ -124,20 +124,32 @@ bool TNumberSchemaItem<NumberType>::isValidNumberType(SmartType type) {
   }
 }
 
-template<typename NumberType>
-Errors::eType TNumberSchemaItem<NumberType>::validate(const SmartObject& Object) {
+template <typename InputType, typename OutputType>
+OutputType IntStaticCast(const InputType& value) {
```

Comment:
Since input param is assumed to be int, then there is no need in const&. no need to fix.